mne/io/base.py

```diff
@@ -2447,7 +2447,7 @@ def _start_writing_raw(name, info, sel=None, data_type=FIFF.FIFFT_FLOAT,
     #
     # Annotations
     #
-    if annotations is not None:  # allow saving empty annotations
+    if annotations is not None and len(annotations):  # don't save empty annot
```

Comment:
Are there still conditions under which `annotations is None` here? It seems like there should not be, so this part of the conditional can be removed